app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -0,0 +1,255 @@
+            currentLocale = currentLocale
+        )
+    }
+    val loadBitmap: LoadBitmapUseCase by lazy { DefaultLoadBitmapUseCase(context) }
+    val selectWallpaper: SelectWallpaperUseCase by lazy { DefaultSelectWallpaperUseCase(context.settings(), store) }
+
+    /**
+     * Contract for usecases that initialize the wallpaper feature.
+     */
+    interface InitializeWallpapersUseCase {
+        /**
+         * Start operations that should be down during initialization, like remote metadata
+         * retrieval and determining the currently selected wallpaper.
+         */
+        suspend operator fun invoke()
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal class DefaultInitializeWallpaperUseCase(
+        private val store: AppStore,
+        private val downloader: WallpaperDownloader,
+        private val fileManager: WallpaperFileManager,
+        private val settings: Settings,
+        private val currentLocale: String,
+        private val possibleWallpapers: List<Wallpaper> = allWallpapers,
+    ) : InitializeWallpapersUseCase {
+
+        /**
+         * Downloads the currently available wallpaper metadata from a remote source.
+         * Updates the [store] with that metadata and with the selected wallpaper found in storage.
+         * Removes any unused promotional or time-limited assets from local storage.
+         * Should usually be called early the app's lifetime to ensure that metadata and thumbnails
+         * are available as soon as they are needed.
+         */
+        override suspend operator fun invoke() {
+            // Quite a bit of code needs to be executed off the main thread in some of this setup.
+            // This should be cleaned up as improvements are made to the storage, file management,
+            // and download utilities.
+            withContext(Dispatchers.IO) {
+                val availableWallpapers = getAvailableWallpapers()
+                val currentWallpaperName = settings.currentWallpaper
+                val currentWallpaper = possibleWallpapers.find { it.name == currentWallpaperName }
+                    ?: fileManager.lookupExpiredWallpaper(currentWallpaperName)
+                    ?: Wallpaper.Default
+
+                fileManager.clean(
+                    currentWallpaper,
+                    possibleWallpapers.filterIsInstance<Wallpaper.Remote>()
+                )
+                downloadAllRemoteWallpapers(availableWallpapers)
+                store.dispatch(AppAction.WallpaperAction.UpdateAvailableWallpapers(availableWallpapers))
+                store.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(currentWallpaper))
+            }
+        }
+
+        private fun getAvailableWallpapers() = possibleWallpapers
+            .filter { !it.isExpired() && it.isAvailableInLocale() }
+
+        private suspend fun downloadAllRemoteWallpapers(availableWallpapers: List<Wallpaper>) {
```

Comment:
why use a parameter + a loop + filter instead of just using `remoteWallpapers` in the companion object?

Reply:
To be able to manipulate the wallpaper list under test, so we can control for things like promotions and expirations. In theory (and as was the case in Turning Red), wallpapers may be locale-restricted and can also separately have their availability expire (this is planned for upcoming seasonal wallpapers). 

Explicitly defining them like they are in the companion object will be going away shortly in any case, as we will be fetching all the wallpapers from metadata stored remotely. The wallpaper type hierarchy may experience some changes as part of that as well.